transport/src/main/java/io/zeebe/transport/impl/util/SocketUtil.java

```diff
@@ -40,7 +40,18 @@
     final int testForkNumber = getTestForkNumber();
     final int testMavenId = getTestMavenId();
 
-    final int min = BASE_PORT + testForkNumber * RANGE_SIZE + (testMavenId * RANGE_SIZE * 10);
+    LOG.info(
+        "Starting socket assignment with testForkNumber {} and testMavenId {}",
+        testForkNumber,
+        testMavenId);
+
+    // ensure limits to stay in available port range
+    assert testForkNumber < 39 : "System property test fork number has to be smaller then 39";
+    assert testMavenId < 10 : "System property test maven id has to be smaller then 10";
+
+    final int testOffset = testForkNumber * 10 + testMavenId;
+
+    final int min = BASE_PORT + testOffset * RANGE_SIZE;
     final int max = min + RANGE_SIZE;
     TEST_FORK_NUMBER = testForkNumber;
     PORT_RANGE = new PortRange(DEFAULT_HOST, min, max);
```

Comment:
Might be enough to give the `RANGE_SIZE`into the ctor, don't need to calculate the max.